packages/rocketchat-lib/server/functions/getFullUserData.js

```diff
@@ -23,13 +23,30 @@ RocketChat.getFullUserData = function({userId, filter, limit}) {
 			services: 1,
 			requirePasswordChange: 1,
 			requirePasswordChangeReason: 1,
-			roles: 1,
-			customFields: 1
+			roles: 1
 		});
 	} else if (limit !== 0) {
 		limit = 1;
 	}
 
+	const metaCustomFields = RocketChat.settings.get('Accounts_CustomFields').trim();
+	let sCustomFields;
+	try {
+		sCustomFields = JSON.parse(metaCustomFields);
+
+		if (sCustomFields) {
+			_.each(sCustomFields, (el, key) => {
+				if (el.public || RocketChat.authz.hasPermission(userId, 'view-full-other-user-info')) {
+					fields[`customFields.${ key }`] = 1;
+				}
+			});
+		}
+	} catch (e) {
+		if (metaCustomFields !== '') {
```

Comment:
Instead of checking this here, let's check it first before we try to parse the json. This way the error being thrown is a valid error instead of a error that will happen on the servers which don't use custom fields. As a server might have setup error handling/monitoring and adding a false positive will not be the best option, in my opinion. :)